atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -293,12 +297,14 @@ static LockAndTimestampServices createRawServices(
             Environment env,
             Supplier<RemoteLockService> lock,
             Supplier<TimestampService> time,
+            Supplier<TimestampStoreInvalidator> invalidator,
             String userAgent) {
         if (config.leader().isPresent()) {
             return createRawLeaderServices(config.leader().get(), env, lock, time, userAgent);
         } else if (config.timestamp().isPresent() && config.lock().isPresent()) {
             return createRawRemoteServices(config, userAgent);
         } else if (config.timelock().isPresent()) {
```

Comment:
Yep, the TM.createWithTimelock() sounds like a good idea. Tracking separately at #1595